src/test/functional/functional-tests.js

```diff
@@ -1005,12 +987,12 @@ describe('functional tests', function() {
     })
 
     step(`removeObject(bucketName, objectName, callback)__Remove ${listObjectsNum} objects`, done => {
-      async.mapLimit(
-        listPrefixArray,
-        10,
-        (objectName, cb) => client.removeObject(bucketName, objectName, cb),
-        done
-      )
+      for(var i = 0; i < listObjectsNum; i++) {
+        client.removeObject(bucketName, listPrefixArray[i], function(e) {
+          if (e) return done(e)
+        })
+      }
+      done()
```

Comment:
this is not correct @kannappanr note that in JS things are async
i.e done() gets called even before any of the objects get removed

Reply:
Reverted the changes with the parallel limit set to 5.